### PR TITLE
Enable Native AOT testing in System.Text.Json.

### LINKED ISSUE #73431: Get System.Text.Json.SourceGeneration tests running with NativeAot
It would be good to go over these failures to see if #73124 is really the only AOT issue.

Full run log is here:
[jsonfail.txt](https://github.com/dotnet/runtime/files/9265839/jsonfail.txt)

The main buckets:

* xUnit doing its thing. If a Fact/Theory is a generic method, xUnit will MakeGeneric it at runtime. This tends to fail. Can be worked around with RD.XML. Not a customer issue.
* Enum converter. I believe this is #73124
    ```
    [FAIL] System.Text.Json.SourceGeneration.Tests.JsonSerializerContextTests.SupportsGenericParameterWithCustomConverterFactory
    System.Reflection.MissingMetadataException : 'System.Text.Json.Serialization.Converters.EnumConverter<System.Text.Json.SourceGeneration.Tests.JsonSerializerContextTests.TestEnum>' is missing metadata. For more information, please visit http://go.microsoft.com/fwlink/?LinkID=392859
      at System.Reflection.Runtime.TypeInfos.RuntimeTypeInfo.get_TypeHandle() + 0x8e
      at System.Reflection.Runtime.MethodInfos.RuntimePlainConstructorInfo`1.Invoke(BindingFlags, Binder, Object[], CultureInfo) + 0x2f
       at System.ActivatorImplementation.CreateInstance(Type, BindingFlags, Binder, Object[], CultureInfo, Object[]) + 0x256
       at System.Text.Json.Serialization.Converters.EnumConverterFactory.Create(Type, EnumConverterOptions, JsonNamingPolicy, JsonSerializerOptions) + 0x8c
       at System.Text.Json.SourceGeneration.Tests.JsonSerializerContextTests.GenericParameterWithCustomConverterFactoryContext.GetConverterFromFactory(JsonSerializerOptions, Type, JsonConverterFactory) + 0x27
       at System.Text.Json.SourceGeneration.Tests.JsonSerializerContextTests.GenericParameterWithCustomConverterFactoryContext.Create_TestEnum(JsonSerializerOptions) + 0x81
       at System.Text.Json.SourceGeneration.Tests.JsonSerializerContextTests.GenericParameterWithCustomConverterFactoryContext.ListTestEnumSerializeHandler(Utf8JsonWriter, List`1) + 0x8f
       at System.Text.Json.JsonSerializer.WriteCore[TValue](Utf8JsonWriter, TValue&, JsonTypeInfo`1) + 0x6e
       at System.Text.Json.JsonSerializer.WriteString[TValue](TValue&, JsonTypeInfo`1) + 0xe0
       at System.Text.Json.JsonSerializer.Serialize[TValue](TValue, JsonTypeInfo`1) + 0x3c
    ```
* Not sure what this is but looks different:
    ```
    [FAIL] System.Text.Json.SourceGeneration.Tests.SerializationContextTests.RoundtripWithCustomConverterProperty_Struct
    System.Reflection.MissingRuntimeArtifactException : MakeGenericMethod() cannot create this generic method instantiation because no code was generated for it: 
    'System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.CreateReflectionJsonTypeInfo<System.Text.Json.SourceGeneration.Tests.StructWithCustomConverterProperty>(System.Text.Json.JsonSerializerOptions)'.
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.GetUncachedMethodInvoker(RuntimeTypeInfo[], MemberInfo) + 0x2f
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.MakeGenericMethod(Type[]) + 0x201
   at System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.CreateJsonTypeInfo(Type, JsonSerializerOptions) + 0x8e
   at System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.GetTypeInfo(Type, JsonSerializerOptions) + 0x47
   at System.Text.Json.JsonSerializerOptions.GetTypeInfoNoCaching(Type) + 0x2a
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey, Func`2) + 0x85
   at System.Text.Json.JsonSerializerOptions.GetTypeInfoInternal(Type, Boolean, Boolean) + 0x35
   at System.Text.Json.JsonSerializer.GetTypeInfo(JsonSerializerOptions, Type) + 0x71
   at System.Text.Json.SourceGeneration.Tests.RealWorldContextTests.RoundtripWithCustomConverterProperty_Struct() + 0x10d
    ```
* This also looks different:
    ```
  [FAIL] System.Text.Json.SourceGeneration.Tests.SerializationContextTests.RoundtripWithCustomConverterProperty_Class
  System.Reflection.MissingRuntimeArtifactException : MakeGenericMethod() cannot create this generic method instantiation because no code was generated for it: 'System.Text.Json.Serialization.Metadata.JsonTypeInfo.CreateJsonPropertyInfo<System.Int32>(System.Text.Json.Serialization.Metadata.JsonTypeInfo,System.Text.Json.JsonSerializerOptions)'.
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.GetUncachedMethodInvoker(RuntimeTypeInfo[], MemberInfo) + 0x2f
   at System.Reflection.Runtime.MethodInfos.RuntimeNamedMethodInfo`1.MakeGenericMethod(Type[]) + 0x201
   at System.Text.Json.Serialization.Metadata.JsonTypeInfo.CreatePropertyUsingReflection(Type) + 0xdf
   at System.Text.Json.Serialization.Metadata.ReflectionJsonTypeInfo`1.CreateProperty(Type, MemberInfo, JsonSerializerOptions, Boolean) + 0xa4
   at System.Text.Json.Serialization.Metadata.ReflectionJsonTypeInfo`1.AddPropertiesAndParametersUsingReflection() + 0x1fa
   at System.Text.Json.Serialization.Metadata.ReflectionJsonTypeInfo`1..ctor(JsonConverter, JsonSerializerOptions) + 0x8c
   at System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.CreateReflectionJsonTypeInfo[T](JsonSerializerOptions) + 0x48

    ```

I haven't really gone through everything.

Going to block this all on this bug. There's little point testing S.T.Json with NativeAOT if it doesn't work.

### LINKED ISSUE #86973: System.Text.Json xunit generic theories not working in Native AOT.
Creating this issue to track the tests disabled to address #73431.

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Fixes #73431. Fixes #86973.

### CODE REVIEW

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs [hunk 1]
```diff
@@ -624,13 +624,13 @@ public async Task ReadISetTOfHashSetT()
 
             if (result.First().Contains(1))
             {
-                Assert.Equal(new HashSet<int> { 1, 2 }, result.First());
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs:627]: Apparently the xunit comparer for sets was using unsupported reflection, so I changed it to something equivalent.

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs [hunk 2]
```diff
@@ -16,6 +16,7 @@ public abstract partial class CollectionTests
     {
         [Theory]
         [MemberData(nameof(GetAsyncEnumerableSources))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/86973", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
Comment (MichalStrehovsky) [inline src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs:19]: This is fixable - add this to the project file:

https://github.com/dotnet/runtime/blob/349bbe938c420af98909c266b53092dd3ae15e13/src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj#L26-L28

Then create the XML file and put the required generic instantiations in it, like this:

https://github.com/dotnet/runtime/blob/ed2634403de655ba68230ad531e2b7cb1c21b87f/src/libraries/System.Runtime/tests/default.rd.xml#L45-L88

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs [hunk 3]
```diff
@@ -624,13 +624,13 @@ public async Task ReadISetTOfHashSetT()
...
-                Assert.Equal(new HashSet<int> { 3, 4 }, result.Last());
+                Assert.True(result.First().SetEquals(new []{ 1, 2 }));
```
Comment (MichalStrehovsky) [inline src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Generic.Read.cs:627]: If you paste the stack I can point you to the RD.XML to fix this. I've seen this, it's fixable.

The problem will also go away once we ingest AOT compatible xunit asserts (https://github.com/dotnet/arcade/pull/13241).

Maybe it would be preferable to fix this with RD.XML to keep good diagnostics for failures.

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs [hunk 4]
```diff
@@ -520,6 +520,7 @@ public void ExtensionProperty_SupportsWritingToCustomSerializerWithExplicitConve
         [InlineData(typeof(Dictionary<string, JsonElement>), typeof(JsonElementOverflowConverter), typeof(JsonElement))]
         [InlineData(typeof(CustomOverflowDictionary<object>), typeof(CustomObjectDictionaryOverflowConverter), typeof(object))]
         [InlineData(typeof(CustomOverflowDictionary<JsonElement>), typeof(CustomJsonElementDictionaryOverflowConverter), typeof(JsonElement))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/86973", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
Comment (MichalStrehovsky) [inline src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:523]: This doesn't look like a generic theory - what's the issue?

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs [hunk 5]
```diff
@@ -559,6 +560,7 @@ ClassWithExtensionData<TDictionary> obj
         [InlineData(typeof(ClassWithJsonElementExtensionDataWithAttributedConverter), typeof(Dictionary<string, JsonElement>), typeof(JsonElement))]
         [InlineData(typeof(ClassWithCustomElementExtensionDataWithAttributedConverter), typeof(CustomOverflowDictionary<object>), typeof(object))]
         [InlineData(typeof(ClassWithCustomJsonElementExtensionDataWithAttributedConverter), typeof(CustomOverflowDictionary<JsonElement>), typeof(JsonElement))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/86973", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
Comment (MichalStrehovsky) [inline src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:563]: Same question

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs (hunk 4 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:523]: The test is running `MethodInfo.MakeGenericInfo()` with the type arguments. Not quite a generic theory but same pattern in principle.

File: src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs [hunk 6]
```diff
@@ -461,8 +461,12 @@ public void Verify()
             {
                 if (data is JsonElement element)
                 {
-                    SimpleTestClass obj = JsonSerializer.Deserialize<SimpleTestClass>(element.GetRawText());
-                    obj.Verify();
+                    if (!PlatformDetection.IsNativeAot)
+                    {
+                        // The reflection-based validation baseline is not available in Native AOT
```
[the remark this answers is not in the document]
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs:466]: Not to my knowledge at runtime.

[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs:466]: Is this about #83844?

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs (hunk 4 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:523]: We probably just need a line in RD.XML to fix this - the problematic one is the JsonElement. Reference types will Just Work. Valuetypes require runtime codegen.

File: src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs:466]: I wouldn't want to enable `IsReflectionEnabledByDefault` in the test project since it would remove the ability to run the validation in CoreCLR.

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs (hunk 2 above)
Comment (MichalStrehovsky) [reply src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs:19]: Up to you but this is literally what we did in every test that we're running. We don't have any test in the product disabled because of this.

File: src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs:466]: Can we condition this codepath on that appcontext switch if that is the problem though?

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs (hunk 2 above)
Comment (jkotas) [reply src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs:19]: https://github.com/dotnet/runtime/pull/83933 has been working towards fixing this. The problem is in xunit.

File: src/libraries/System.Text.Json/tests/Common/JsonCreationHandlingTests.Generic.cs [hunk 7]
```diff
@@ -42,8 +42,15 @@ internal struct StructList<T> : IList<T>, IList
             set => _list[index] = (T)value;
         }
 
+        [JsonConstructor]
```
Comment (eerhardt) [inline src/libraries/System.Text.Json/tests/Common/JsonCreationHandlingTests.Generic.cs:45]: Are we affecting test coverage in anyway by adding this attribute and the 2nd constructor? Basically, are we losing coverage for the case when the struct just has an unannotated parameterless ctor?

File: src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Serialize.cs [hunk 8]
```diff
@@ -30,6 +30,7 @@ public class Employee
         }
 
         [Fact]
+        [ActiveIssue("Newtonsoft.Json baseline not supported in Native AOT", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
Comment (eerhardt) [inline src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Serialize.cs:33]: Should we make this a ConditionalFact instead? This isn't really an "ActiveIssue". Newtonsoft.Json is not supported, intentionally. So it isn't really an "issue".

File: src/libraries/System.Text.Json/tests/Common/JsonCreationHandlingTests.Generic.cs (hunk 7 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/JsonCreationHandlingTests.Generic.cs:45]: It shouldn't, the particular suite tests the populate semantics of the containing type and does attempt to deserialize the collection itself. I added the annotation out of abundance of caution, so that the newly added constructor doesn't impact the serialization semantics of the collection type itself.

File: src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Serialize.cs (hunk 8 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Serialize.cs:33]: Good question, I tried using it but I couldn't find a way to condition it on the negation of a boolean. Guess I need to add a `IsNotNativeAot` property?

Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Serialize.cs:33]: Ah, looks like somebody has already added that :-)

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs (hunk 2 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs]: Pushed new changes that should enable testing for all remaining generic theories.

File: src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/TestClasses/TestClasses.cs]: I changed it so that it uses source generation as the baseline.

File: src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs [hunk 9]
```diff
@@ -48,37 +46,10 @@ public async Task WriteNestedAsyncEnumerable<TElement>(IEnumerable<TElement> sou
...
-            await StreamingSerializer.SerializeWrapper(stream, new AsyncEnumerableDto<TElement> { Data = asyncEnumerable }, options);
-
-            JsonTestHelper.AssertJsonEqual(expectedJson, stream.AsString());
-            Assert.Equal(1, asyncEnumerable.TotalCreatedEnumerators);
-            Assert.Equal(1, asyncEnumerable.TotalDisposedEnumerators);
-        }
-
-        [Theory]
-        [MemberData(nameof(GetAsyncEnumerableSources))]
-        public async Task WriteNestedAsyncEnumerable_DTO<TElement>(IEnumerable<TElement> source, int delayInterval, int bufferSize)
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.AsyncEnumerable.cs:69]: This test was identical to the one above, so I deleted it.

File: src/libraries/System.Text.Json/tests/Common/ConstructorTests/ConstructorTests.ParameterMatching.cs [hunk 10]
```diff
@@ -1049,6 +1049,7 @@ public async Task InvalidJsonFails()
             await Assert.ThrowsAsync<JsonException>(() => Serializer.DeserializeWrapper<Point_2D_Struct>("{true"));
         }
 
+#if !BUILDING_SOURCE_GENERATOR_TESTS // Anonymous types not supported in source gen
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/Common/ConstructorTests/ConstructorTests.ParameterMatching.cs:1052]: The test is hardcoding reflection, so I'm just disabling it for source gen.

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs [hunk 11]
```diff
@@ -428,104 +428,93 @@ public override void Write(Utf8JsonWriter writer, CustomOverflowDictionary<JsonE
...
-            typeof(ExtensionDataTests)
-                .GetMethod(nameof(ExtensionProperty_SupportsWritingToCustomSerializerWithOptionsInternal), BindingFlags.Static | BindingFlags.NonPublic)
-                .MakeGenericMethod(overflowType, converterType)
-                .Invoke(null, null);
-        }
-
-        private static void ExtensionProperty_SupportsWritingToCustomSerializerWithOptionsInternal<TDictionary, TConverter>()
-            where TDictionary : new()
-            where TConverter : JsonConverter, new()
+        [MemberData(nameof(GetCustomOverflowConverters))]
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:431]: Refactored the tests somewhat to reduce the number of generic parameters needed to drive the tests.

File: src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn3.11.rd.xml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+<Directives>
+  <Application>
+    <!-- NB Shared specializations must also be replicated in the roslyn4.4.rd.xml file -->
```
Comment (MichalStrehovsky) [inline src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn3.11.rd.xml:3]: I assume we'll need to copy/paste the contents of the other XML here to get a successful test pass?

**Review by MichalStrehovsky — COMMENTED** [review]
Looks good to me from NativeAOT perspective, but can't speak about the test changes in general. Thank you!

File: src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn3.11.rd.xml (hunk 12 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn3.11.rd.xml:3]: The Roslyn3.11 project only runs a subset of tests, so specializations are not necessary. Left an RD file as a placeholder in case that changes in the future.

File: src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml [hunk 13]
```diff
@@ -0,0 +1,56 @@
+<Directives>
+  <Application>
+    <!-- NB Shared specializations must also be replicated in the roslyn3.11.rd.xml file -->
+    <Assembly Name="System.Text.Json.SourceGeneration.Roslyn4.4.Tests">
```
Comment (krwq) [inline src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml:4]: why are those needed? Do we know the reason they're trimmed out?

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs [hunk 14]
```diff
@@ -428,104 +428,93 @@ public override void Write(Utf8JsonWriter writer, CustomOverflowDictionary<JsonE
...
+            var options = Serializer.CreateOptions(makeReadOnly: false);
+            options.Converters.Add(dictionaryConverter);
 
-            string json = JsonSerializer.Serialize(root, options);
+            string json = await Serializer.SerializeWrapper(root, options);
             Assert.Equal(@"{""MyCustomOverflowWrite"":""OverflowValueWrite""}", json);
         }
 
-        private interface IClassWithOverflow<T>
+        public interface IClassWithOverflow
```
Comment (krwq) [inline src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:447]: out of curiosity, does the visibility change have influence on trimming or are we just reusing this class elsewhere?

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs [hunk 15]
```diff
@@ -428,104 +428,93 @@ public override void Write(Utf8JsonWriter writer, CustomOverflowDictionary<JsonE
...
         {
-            public T Overflow { get; set; }
+            public IDictionary GetOverflow();
```
Comment (krwq) [inline src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:449]: what's the reason for both method and property?

**Review by krwq — APPROVED** [review]
LGTM, thanks!

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs (hunk 15 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:449]: To avoid the helper being captured in the serialization contract for the interface type.

File: src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs (hunk 14 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs:447]: It is passed as a parameter in theory method, so the type needs to be public.

File: src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml (hunk 13 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml:4]: They are helping the AOT compiler generate generic method specializations we know will be needed at runtime (but can't be inferred at compile time).

Comment (krwq) [reply src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml:4]: do we know why they can't be inferred?

Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml:4]: It's the usual issue of trying to do `Type.MakeGenericType`/`Method.MakeGenericMethod` in AOT. We're working on a fix specifically for theories though: https://github.com/dotnet/runtime/pull/86975#discussion_r1212381894
